libPhoneNumber/NBGeocoderMetadataHelper.m

```diff
@@ -0,0 +1,132 @@
+//
+//  NBGeocoderMetadataHelper.m
+//  libPhoneNumberiOS
+//
+//  Created by Rastaar Haghi on 6/12/20.
+//  Copyright © 2020 ohtalk.me. All rights reserved.
+//
+
+#import "NBGeocoderMetadataHelper.h"
+
+@implementation NBGeocoderMetadataHelper {
+ @private
+  NSString *databasePath;
+  sqlite3 *DB;
+  sqlite3_stmt *selectStatement;
+}
+
+- (instancetype)initWithCountryCode:(NSNumber *)countryCode withLanguage:(NSString *)language {
+  self = [super init];
+
+  self.countryCode = countryCode;
+  self.language = language;
+  // grab bundle of current pod instance
+  NSBundle *bundle = [NSBundle bundleForClass:self.classForCoder];
+  NSURL *bundleURL = [[bundle resourceURL] URLByAppendingPathComponent:@"Resources.bundle"];
+  // create string for database directory
+  NSString *databasePath = [NSString stringWithFormat:@"%@%@.db", bundleURL, self.language];
+  self->databasePath = databasePath;
+
+  // open the database position
+  sqlite3_open([databasePath UTF8String], &self -> DB);
+
+  // set up prepared statements for inserting entry and searching for entry
+  sqlite3_prepare_v2(
+      self->DB,
+      [[NSString
+          stringWithFormat:
+              @"WITH RECURSIVE cnt(x) AS ( SELECT 1 UNION ALL SELECT x+1 FROM cnt LIMIT "
+              @"length(?)), toSearch as (SELECT substr(?, 1, x) as indata FROM cnt) select "
+              @"nationalnumber, description, length(nationalnumber) as natLength from "
+              @"geocodingPairs%@ where NATIONALNUMBER in toSearch order by natLength desc limit 1",
+              countryCode] UTF8String],
+      -1, &self->selectStatement, NULL);
+  return self;
+}
+
+- (void)dealloc {
+  sqlite3_finalize(self->selectStatement);
+  sqlite3_close_v2(self->DB);
+}
+
+// returns a select statement for finding description given a phone number
+- (int)createSelectStatement:(NBPhoneNumber *)phoneNumber {
+  int sqliteResultCode;
+  @autoreleasepool {
+    sqliteResultCode = sqlite3_reset(self->selectStatement);
```

Comment:
Can you also call sqlite3_clear_bindings() too? Since sqlite3_reset() doesn't clear the bindings.

Reply:
Done.